### PR TITLE
Fix #627 [Model Editor] Panel Editor - add Parameter and add Element Definition improvements

### LINKED ISSUE #627: [Model Editor] Panel Editor - add Parameter and add Element Definition improvements
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of the COMET-WEB
- [x] I have searched [open](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/issues) and [closed](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
The following improvements need to be implemented when performing **Add Parameter** and Add **Element Definition**

Add Parameter:
   - [x] the Owner combox shall list the domains of expertise in alphabetical order. Only the active domains of the engineeringmodelsetup shall be displayed
   - [x] All the ParameterTypes present in the chain of rdls ahall be available. Currently it seems only number of items is present. I case a quantitykind is selected, the user also needs to select a scale. Show a new combox where the default scale is pre-selected.
   - [x] The ParameterTypes combox needs to show name and shortname: name [shortname], the user needs to be able to start typing the name for easy selection;  The parametertypes shall be sorted in alphabetical order; The same applies to the measurementscale selection.

Add Element Definition:
   - [x] the Owner combox shall list the domains of expertise in alphabetical order. Only the active domains of the engineeringmodelsetup shall be displayed
   - [x] When the dialog is popped multiple times and cancelled, duplicate categories are displayed (it seems the same form is re-used and not all properties are cleared when cancelled)

### System Configuration

- COMET version:
  - [x] COMET-WEB: 4.4.0

### PR DESCRIPTION
**Pull request by joao4all**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/pulls) open
- [x] I have verified that I am following the COMET-WEB [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/COMET-WEB-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description
Closes #627 [Model Editor] Panel Editor - add Parameter and add Element Definition improvements

### CODE REVIEW

File: COMET.Web.Common/Components/Selectors/MeasurementScaleSelector.razor [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+<DxComboBox Data="@(this.ViewModel.AvailableMeasurementScales.Select(x => new SelectorModelBaseWrapper<MeasurementScale>(x)))"
+            TData="SelectorModelBaseWrapper<MeasurementScale>"
+            TValue="MeasurementScale"
+            @bind-Value="@(this.ViewModel.SelectedMeasurementScale)"
+            FilteringMode="DataGridFilteringMode.Contains"
+            ClearButtonDisplayMode="DataEditorClearButtonDisplayMode.Auto"
+            TextFieldName="@(nameof(SelectorModelBaseWrapper<ParameterType>.DisplayText))"
+            ValueFieldName="@(nameof(SelectorModelBaseWrapper<ParameterType>.WrappedThing))"
+            NullText="Select a Parameter Type"/>
```
Comment (antoineatstariongroup) [inline COMET.Web.Common/Components/Selectors/MeasurementScaleSelector.razor:35]: The text should be Select a Scale

File: COMET.Web.Common/Components/Selectors/MeasurementScaleSelector.razor.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        /// Gets or sets the <see cref="IMeasurementScaleSelectorViewModel" />
+        /// </summary>
+        [Parameter]
+        public IMeasurementScaleSelectorViewModel ViewModel { get; set; }
+
+        /// <summary>
+        /// Text to be displayed when the selector is shown
+        /// </summary>
+        [Parameter]
+        public string DisplayText { get; set; } = "Filter on Parameter Type:";
```
Comment (antoineatstariongroup) [inline COMET.Web.Common/Components/Selectors/MeasurementScaleSelector.razor.cs:48]: Why "Filter on Parameter Type" for MeasurementScale?

**Review by antoineatstariongroup — CHANGES_REQUESTED** [review]
Also, do you remove ParameterType that are already present in the ElementDefinition? It is not allowed to have duplicates

File: COMET.Web.Common/Components/Selectors/MeasurementScaleSelector.razor.cs (hunk 2 above)
Comment (joao4all) [reply COMET.Web.Common/Components/Selectors/MeasurementScaleSelector.razor.cs]: Mb there, forgot to delete the parameter. Fixed in the newest commit :)

**Comment by joao4all** [comment]
> Also, do you remove ParameterType that are already present in the ElementDefinition? It is not allowed to have duplicates

The parameter types that are present in the ElementDefinition are not displayed for adding.

**Review by antoineatstariongroup — APPROVED** [review]

**Pull request merged**